### PR TITLE
gpl: Fix finding and replacing pins in clustering

### PR DESCRIPTION
**Pull request by therealshreyas**
The following changes have been made in this PR:
(1) Added test_cell to asap7. Currently, asap7 lacks test_cell, which means that the scan pins are not understood by sta. I added test_cell to src/gpl/test/SingleBitFolder/.
(2) Added updated 2- and 4-bit MBFF trays (courtesy of Mr. Dooseok Yoon). These files are already in ORFS, but don't seem to be in OR yet.
(3) Updated mbff.cpp to fetch scan, supply and set pins. This way, the netlist is actually modified as expected.
(4) Updated mbff.cpp to fetch launch-capture FF-pairs (added top_k parameter to the tcl script, default = 0).
(5) Fixed GetSlots() so that slots are fetched w.r.t. tray center (rather than bottom-left corner). 
(6) Updated power ratio calculation (previously hard coded).
(7) Updated ILP to reflect recent experiments. 
(8) Added checks to make sure that FFs and MBFFs are valid.
(9) Added a function equivalence check that works regardless of LibertyPort names. Possible todo: save function traversal for runtime optimization (the current implementation should work fine unless the number of distinct tray masters is massive). 
(10) Fixed bug where set pins were found with function equivalence (this logic is incorrect). Updated to use seq->clear() and seq->preset().

### CODE REVIEW

**Comment by maliberty** [comment]
Please write a summary of this change ('fix' is too brief).

**Comment by therealshreyas** [comment]
> Please write a summary of this change ('fix' is too brief).

Done!

**Comment by maliberty** [comment]
Have you run asap7/aes-mbff with this change?

**Review by maliberty — CHANGES_REQUESTED** [review]

File: src/gpl/test/SingleBit/asap7sc7p5t_SEQ_LVT_TT_nldm_220123.lib
[the remark this answers is not in the document]
Comment (therealshreyas) [reply src/gpl/test/SingleBit/asap7sc7p5t_SEQ_LVT_TT_nldm_220123.lib:1]: It has test_cell (+ I need LVT for sensible power ratio calculation)

File: src/gpl/test/SingleBit/asap7sc7p5t_SEQ_RVT_FF_nldm_220123.lib
[the remark this answers is not in the document]
Comment (therealshreyas) [reply src/gpl/test/SingleBit/asap7sc7p5t_SEQ_RVT_FF_nldm_220123.lib:1]: It has test_cell

File: src/gpl/test/clust03.tcl
[the remark this answers is not in the document]
Comment (therealshreyas) [reply src/gpl/test/clust03.tcl:1]: clust01 and clust02 provide enough coverage (varying alpha, split_size, and input size)

File: src/gpl/test/SingleBit/asap7sc7p5t_28_L_1x_220121a.lef
[the remark this answers is not in the document]
Comment (therealshreyas) [reply src/gpl/test/SingleBit/asap7sc7p5t_28_L_1x_220121a.lef:1]: No (R / L difference)

**Review by maliberty — APPROVED** [review]

**Pull request merged**